app/services/LegacyIdentityDbUserService.scala

```diff
@@ -1,78 +1,52 @@
 package services
 
-import model.{Address, Name, Permission, PhoneNumber, User}
-import play.api.libs.json.{JsValue, Json, Reads}
+import model.Permission
+import play.api.Logging
+import play.api.libs.json.{JsArray, JsValue, Json, Reads}
 import slick.basic.DatabaseConfig
 import slick.jdbc.JdbcProfile
-import slick.lifted.TableQuery
 
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Failure
+import scala.util.chaining.*
 
-class LegacyIdentityDbUserService(val dbConfig: DatabaseConfig[JdbcProfile])(implicit ec: ExecutionContext)
-    extends UserService {
+class LegacyIdentityDbUserService(val dbConfig: DatabaseConfig[JdbcProfile])(implicit ctx: ExecutionContext)
+    extends LegacyUserService
+    with Logging {
 
   import dbConfig.profile.api.*
 
   private val db = dbConfig.db
 
-  private class UserTable(tag: Tag) extends Table[UserRow](tag, "users") {
-    def identityId = column[String]("id")
-    def brazeId = column[Option[String]]("braze_uuid")
-    def jdoc = column[String]("jdoc")
-
-    def * = (identityId, brazeId, jdoc).mapTo[UserRow]
-  }
-
-  private val users = TableQuery[UserTable]
-
-  override def healthCheck(): Future[Unit] =
-    db.run(sql"SELECT 1".as[Int]).map(_ => ())
-
-  def fetchUserByIdentityId(id: String): Future[Option[User]] = {
-    val action = users.filter(_.identityId === id).take(1).result
-    val result = db.run(action)
-    result.map(_.map(UserRow.toUser).headOption)
+  def healthCheck(): Future[Unit] =
+    db.run(sql"SELECT 1".as[Int])
+      .map(_ => ())
+      .tap(_.onComplete {
+        case Failure(exception) => logger.error(s"Health check failed: ${exception.getMessage}")
+        case _                  => ()
+      })
+
+  def fetchByIdentityId(id: String): Future[Option[LegacyUser]] =
```

Comment:
This might be overkill but do we want to drop down to pull more IDAPI fields if Okta doesn't respond correctly? I suppose our source of truth should be Okta, not IDAPI, so this is corect.

Reply:
In principle, what's in Okta should be more reliable than what's in Postgres as it gets updated first so if there's something missing in Okta we should probably assume it's meant to be missing.